lib/clAdapters/google/errors.ts

```diff
@@ -0,0 +1,46 @@
+// Type union instead of enum so calling code can get name without type details
+export type GoogleErrorType =
+  'InvalidGrant' |
+  'UnauthorizedClient'
+;
+
+export interface InvalidGrant {
+  kind: 'InvalidGrant';
+  err: Error;
+}
+
+export interface UnauthorizedClient {
+  kind: 'UnauthorizedClient';
+  err: Error;
+}
+
+export type GoogleError = InvalidGrant | UnauthorizedClient;
+
+export function createGoogleError<T extends GoogleErrorType>(
+  kind: T,
+  err?: Error
+): GoogleError {
+  if (!err) {
+    err = new Error();
+    Error.captureStackTrace(err, createGoogleError);
+  }
+
+  // return {
+  //     kind,
+  //     err
+  // };
+  // Type value is not enforceable:
```

Comment:
try hovering over`createError` in `createError('invalid')` to see the return type